src/popup/components/BalanceInfo.vue

```diff
@@ -3,34 +3,48 @@
     class="balance-info"
     data-cy="balance-info"
   >
-    <AeBalance :balance="Number(balances[idx])" />
+    <AeBalance :balance="balance" />
     <div class="display-value">
-      {{ formatCurrency(balances[idx] * currentCurrencyRate) }}
+      {{ formattedCurrency }}
     </div>
   </div>
 </template>
 
-<script>
-import { pick } from 'lodash-es';
-import { mapGetters, mapState } from 'vuex';
+<script lang="ts">
+import { computed, defineComponent } from '@vue/composition-api';
+import BigNumber from 'bignumber.js';
+import { useGetter, useState } from '../../composables';
+import { rxJsObservableToVueState } from '../utils';
 import AeBalance from './AeBalance.vue';
 
-export default {
-  components: { AeBalance },
+export default defineComponent({
+  components: {
+    AeBalance,
+  },
   props: {
     accountIdx: { type: Number, default: -1 },
   },
-  subscriptions() {
-    return pick(this.$store.state.observables, ['balances']);
-  },
-  computed: {
-    ...mapState('accounts', ['activeIdx']),
-    ...mapGetters(['formatCurrency', 'currentCurrencyRate', 'accounts']),
-    idx() {
-      return this.accountIdx === -1 ? this.activeIdx : this.accountIdx;
-    },
+  setup(props, { root }) {
+    const activeIdx = useState('accounts', 'activeIdx');
+    const formatCurrency = useGetter('formatCurrency');
+    const currentCurrencyRate = useGetter('currentCurrencyRate');
+
+    const balances = rxJsObservableToVueState<BigNumber[]>(
+      (root.$store.state as any).observables.balances,
+    );
+
+    const idx = computed(() => props.accountIdx === -1 ? activeIdx.value : props.accountIdx);
+    const balance = computed(() => balances.value[idx.value].toNumber());
```

Comment:
If I'm not mistaken we do have a `balance` state in observables

Reply:
Yes, I thought about the same but this component is not showing the balance of the current account but the balance based on the IDX. So we are picking the balance from the list.